### PR TITLE
Clean up lint errors in several Python scripts

### PR DESCRIPTION
**Pull request by miguelsousa**

### CODE REVIEW

File: afdko/Tools/SharedData/FDKScripts/ConvertFontToCID.py [hunk 1]
```diff
@@ -35,871 +34,1007 @@
...
-		raise FontInfoParseError("Error: Failed finding FontBBox in tx log from  %s." % (fPath))
-	fontBBox = m.group(1).split(",")
-	fontBBox = map(int, fontBBox)
-	return fontBBox
+    fontBBox = [-200, -200, 1000, 100]
+    command = "tx -mtx -2  \"%s\" 2>&1" % fPath
+    data = FDKUtils.runShellCmd(command)
+    if not data:
+        raise FontInfoParseError("Error: Failed getting log from tx from %s, "
+                                 "when tryingg to get FontBBox." % fPath)
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/ConvertFontToCID.py]: too many g's for 'trying'

File: afdko/Tools/SharedData/FDKScripts/ConvertFontToCID.py [hunk 2]
```diff
@@ -35,871 +34,1007 @@
...
 def getFontName(fPath):
-	command = "tx -dump -0 \"%s\" 2>&1" % (fPath)
-	data = FDKUtils.runShellCmd(command)
-	if not data:
-		raise FontInfoParseError("Error: Failed getting  log from tx from %, when tryin to get FontName." % (fPath))
+    command = "tx -dump -0 \"%s\" 2>&1" % fPath
+    data = FDKUtils.runShellCmd(command)
+    if not data:
+        raise FontInfoParseError("Error: Failed getting log from tx from %s, "
+                                 "when tryin to get FontName." % fPath)
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/ConvertFontToCID.py]: not enough g's for 'trying'

File: afdko/Tools/SharedData/FDKScripts/FDKUtils.py [hunk 3]
```diff
@@ -1,181 +1,200 @@
...
+            if output:
+                print(output, end=' ')
+                logList.append(output)
+            if proc.poll():
+                output = proc.stdout.readline()
+                if output:
+                    print(output, end=' ')
+                    logList.append(output)
+                break
+        log = "".join(logList)  # XXX 'log' variable is never used
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/FDKUtils.py]: I think we can go ahead and remove `log` and `logList`.

File: afdko/Tools/SharedData/FDKScripts/ufoTools.py [hunk 4]
```diff
@@ -364,2003 +366,2231 @@ def debugMsg(*args):
...
+        # If the program always reads data from the default layer,
+        # and we have just created a new glyph in the processed layer,
+        # then reset the history.
+        if (not self.useProcessedLayer) and changed:
+            self.hashMap[glyphName] = [srcHash, [self.programName]]
+            return
+        else:
+            try:
+                # XXX programHistoryIndex is assigned but never used
+                programHistoryIndex = historyList.index(self.programName)
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/ufoTools.py]: I suggest replacing this try/except block with:
```python
if self.programName not in historyList:
    historyList.append(self.programName)
```

File: afdko/Tools/SharedData/FDKScripts/ufoTools.py [hunk 5]
```diff
@@ -364,2003 +366,2231 @@ def debugMsg(*args):
...
+    # If dictKey is defined, parse and return only the data for that key.
+    # This helps avoid needing to parse all plist types. We need to support
+    # only data for known keys amd lists.
+    plistDict = {}
+    plistKeys = []
+
+    # We uses this rather than the plistlib in order to get a list that
+    # allows preserving key order.
+    fp = open(filePath, "r")
+    data = fp.read()
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/ufoTools.py]: I think we should remove *all* references to `lastPathOp`.
I count 16 assignments and zero actual usage.

File: afdko/Tools/SharedData/FDKScripts/ConvertFontToCID.py [hunk 6]
```diff
@@ -35,871 +34,1007 @@
...
+                    "Unrecognized token in base state: %s" % token)
+
+        elif state == inValue:
+            # We are processing a list value for a base state token.
+            if token[-1] in ["]", ")"]:
+                # found the last token in the list value.
+                state = baseState
+
+        elif state == inDictValue:
+            # XXX dictValueList & dictKeyWord are not defined!!
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/ConvertFontToCID.py]: I think we should add the following at the top of this method:
```python
dictValueList = []
dictKeyWord = None
```

File: afdko/Tools/SharedData/FDKScripts/ConvertFontToCID.py [hunk 7]
```diff
@@ -35,871 +34,1007 @@
...
+            continue
+        zoneList.sort()
+        # Now check for conflicts.
+        prevZone = zoneList[0]
+        goodZoneList.append(prevZone[1])
+        goodZoneList.append(prevZone[0])
+        zoneBuffer = 2 * prefDDict.BlueFuzz + 1
+        for zone in zoneList[1:]:
+            if (ki == 0) and (len(zoneList) >= 14):
+                # XXX fdDictName is not defined!!
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/ConvertFontToCID.py]: I think we should put the following at the top of this method:
```python
fdDictName = None
```

**Comment by miguelsousa** [comment]
@cjchapman I think I've addressed everything you mentioned.

**Pull request merged**